engine/src/main/java/org/terasology/rendering/dag/nodes/RefractiveReflectiveBlocksNode.java

```diff
@@ -224,10 +229,11 @@ public void process() {
         chunkMaterial.setInt("textureWaterNormalAlt", 4, true);
         chunkMaterial.setInt("textureWaterReflection", 5, true);
         chunkMaterial.setInt("texSceneOpaque", 6, true);
+        chunkMaterial.setInt("texSceneOpaqueDepth", 7, true);
         if (normalMappingIsEnabled) {
-            chunkMaterial.setInt("textureAtlasNormal", 7, true);
+            chunkMaterial.setInt("textureAtlasNormal", 8, true);
             if (parallaxMappingIsEnabled) {
-                chunkMaterial.setInt("textureAtlasHeight", 8, true);
+                chunkMaterial.setInt("textureAtlasHeight", 9, true);
```

Comment:
Quick chip in, since I am responsible for this code: This was done in this (and similar Nodes) because if there exist multiple branches of conditions that might get triggered independently, then texture++ can attach wrong buffers to wrong positions. In this specific case, since all the if conditions are linear and dependent on each other, we can safely convert it, but the same can't be done in some other Nodes.

Reply:
Good point @vampcat. I feel texture slot management is something we could do a lot better as right now textures get assigned and counterassigned to the same texture slot, unnecessarily. 

Theorethically we should leave each texture, as much as possible, in a stable place, with only the least used textures moving around. 

But this is out of the scope of this PR. So, let's leave this part as it is, until we have the time to work on it specifically.

Reply:
Actually, this part of the code will get *much* cleaner as soon as shader features get thrown out of the window, since the only reason we're having to set them in process() right now is ACTIVATE_FEATURE renders the state change that was supposed to set this useless.